app/models/nic/bmc.rb

```diff
@@ -1,7 +1,14 @@
 module Nic
   class BMC < Managed
     PROVIDERS = %w(IPMI)
+    before_validation :ensure_physical
     validates :provider, :presence => true, :inclusion => { :in => PROVIDERS }
+    validates :mac, :presence => true, :if => :managed?
+
+    def virtual
```

Comment:
This is not the root cause, we should not allow to access modify this attribute at all. Nic::BMC can never be virtual but this is outcome of using STI, you always inherit all attributes, even if they are relevant only for some children.

So this way we make sure that we have the value that makes most sense in DB so searching works as expected.